src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2883,16 +2884,16 @@ LogicalResult ONNXTileOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
   // Cannot infer shape if no shape exists.
   if (!input().getType().isa<RankedTensorType>())
-    return emitError("Input tensor not ranked");
+    return success();
 
   // Read 'repeats' value.
   if (!repeats().getType().isa<RankedTensorType>())
-    return emitError("Repeats tensor not ranked");
+    return success();
 
   // 'repeats' tensor is an 1D tensor.
   auto repeatsTensorTy = repeats().getType().cast<RankedTensorType>();
   if (repeatsTensorTy.getShape().size() != 1)
-    return emitError("Repeats tensor must have rank one");
+    return success();
```

Comment:
Same here, must have a rank of 1. How do we keep track of that?

It would seems that we should rather have a 

```
tolerateOrEmitError(phase, message)
```

in which you can switch between generating errors or accepting the current "violation"

Reply:
I put the check back.